### PR TITLE
Dbapi

### PR DESCRIPTION
**Pull request by RaviCompro**
Adding database apis.

### CODE REVIEW

File: server/routes/skillEngineRoutes.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+const skillEngineController = require('../controllers/skillEngineController');
+const databaseFileStoreManager = require('../controllers/databaseFileStoreManager');
+
+/* GET api listing. */
+router.get('/uiconfig/:templateId', (req, res) => {
+    let templateId = req.params.templateId;
+    let data = {};
+    skillEngineController.getUIConfig(req.params.templateId, data, (path) => {
+        
+        res.download(path);
```
Comment (dhirendra-singh81) [inline server/routes/skillEngineRoutes.js]: this should return a json content

File: server/routes/skillEngineRoutes.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    });
+});
+
+router.get('/skillxml/:templateId', (req, res) => {
```
Comment (dhirendra-singh81) [inline server/routes/skillEngineRoutes.js]: don't need skillxml route

File: server/routes/skillEngineRoutes.js [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+    databaseFileStoreManager.getSkillXML(req.params.templateId, (path) => {
+        console.log(path);
+        res.download(path);
+    });
+});
+
+router.get('/iomap/:templateId', (req, res) => {
```
Comment (dhirendra-singh81) [inline server/routes/skillEngineRoutes.js]: remove it

File: server/routes/skillEngineRoutes.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    databaseFileStoreManager.getIOMap(req.params.templateId, (path) => {
+        console.log(path);
+        res.download(path);
+    });
+});
+
+router.get('/skillmodel/:templateId', (req, res) => {
+    databaseFileStoreManager.getSkillModel(req.params.templateId, (path) => {
```
Comment (dhirendra-singh81) [inline server/routes/skillEngineRoutes.js]: rename to skilluimodel

and should return json content

**Review by dhirendra-singh81 — COMMENTED** [review]
Skill File Store structure

skills/<application>/<templateid>/uiconfig.json
skills/<application>/<templateid>/skilluimodel.json 
skills/<application>/<templateid>/skill.xml
skills/<application>/<templateid>/iomap.json

File: server/controllers/databaseAccessLayerController.js [hunk 5]
```diff
@@ -0,0 +1,27 @@
+//const skillController = require('./skillController');
+const skillConfigRepoModel = require('../molels/skillConfigRepoModel').skill_config_repo;
+
+module.exports.getUIConfigPath = function(templateId, callback) {
```
Comment (dhirendra-singh81) [inline server/controllers/databaseAccessLayerController.js]: should only return uiconfig file path

Comment (RaviCompro) [reply server/controllers/databaseAccessLayerController.js]: Code updated to return file specific path.

File: server/routes/skillEngineRoutes.js (hunk 1 above)
Comment (RaviCompro) [reply server/routes/skillEngineRoutes.js]: Code updated to return the JSON content.

File: server/routes/skillEngineRoutes.js (hunk 2 above)
Comment (RaviCompro) [reply server/routes/skillEngineRoutes.js]: Unwanted routes commented out in skillEngineRoutes.js

File: server/routes/skillEngineRoutes.js (hunk 4 above)
Comment (RaviCompro) [reply server/routes/skillEngineRoutes.js]: Code updated to return JSON content.

**Review by dhirendra-singh81 — APPROVED** [review]

**Pull request merged**